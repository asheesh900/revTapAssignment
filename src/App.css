.topContainer {
    width: 100%;
    min-height: 100vh;
    background-color: #dbf2ff;
    background-size: cover;
}