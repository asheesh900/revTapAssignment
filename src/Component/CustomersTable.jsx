import React from 'react'

function CustomersTable({customers}) {
    return (
        <div className = "container my-4">
            <h1 className = "text-center my-4">Customers</h1>
            <table className="table table-striped  table-responsive">
                <thead className = "thead-dark">
                    <tr>
                    <th scope="col">Sr. No.</th>
                    <th scope="col">ID</th>
                    <th scope="col">First Name</th>
                    <th scope="col">Last Name</th>
                    <th scope="col">Email</th>
                    <th scope="col">Created_at</th>
                    <th scope="col">Orders</th>
                    </tr>
                </thead>
                <tbody>
                    {
                       customers && customers.map((customer, i) => {
                            return (
                                <tr key = {customer.id}>
                                    <th>{i+1} </th>
                                    <th>{customer.id} </th>
                                    <th>{customer.firstName} </th>
                                    <th>{customer.lastName} </th>
                                    <th>{customer.email} </th>
                                    <th>{customer.created} </th>
                                    <th>{customer.orders} </th>
                                </tr>
                            )
                        })
                    }
                </tbody>
            </table>
        </div>
    )
    
}

export default CustomersTable
