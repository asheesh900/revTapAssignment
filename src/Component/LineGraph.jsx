import React from 'react';
import {Line} from 'react-chartjs-2';

function LineGraph({totalPricePerDay, date}) {

    const data = {
        labels: date,
        datasets: [
          {
            label: `Orders' Total Price `,
            fill: false,
            lineTension: 0.1,
            backgroundColor: 'red',
            borderColor: 'rgba(75,192,192,1)',
            borderCapStyle: 'butt',
            borderDash: [],
            borderDashOffset: 0.0,
            borderJoinStyle: 'miter',
            pointBorderColor: 'rgba(75,192,192,1)',
            pointBackgroundColor: '#fff',
            pointBorderWidth: 1,
            pointHoverRadius: 5,
            pointHoverBackgroundColor: 'rgba(75,192,192,1)',
            pointHoverBorderColor: 'rgba(220,220,220,1)',
            pointHoverBorderWidth: 2,
            pointRadius: 1,
            pointHitRadius: 10,
            data: totalPricePerDay
          }
        ]
      };
    return (
        <div className = "container text-center my-4">
            <h2 className = "my-4">Orders' Total Price Vs Date</h2>
            <Line data={data} />
        </div>
    )
}

export default LineGraph
