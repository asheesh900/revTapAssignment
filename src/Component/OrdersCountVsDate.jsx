import React, {useState} from 'react';
import db from '../Data/db.json'
import BarGraph from './BarGraph';

function OredersCountVsDate() {
    let [orders] = useState(db.orders)

    let ordersCopy = [...orders]  // shallow copy

    // sorting orders based on date

    ordersCopy.sort((a, b) => {
        let dateA = a.created.toUpperCase();
        let dateB = b.created.toUpperCase();

        if(dateA < dateB) {
            return -1;
        } else if(dateA > dateB) {
            return 1;
        } else {
            return 0;
        }
    })

    let ordersCopy1 = JSON.parse(JSON.stringify(ordersCopy)) // detaching reference of its blueprint (deep copy)

    // extracting date in readable format

    ordersCopy1 = ordersCopy1.map(ele => {
        ele["created"] = ele["created"].split("T");
        return ele
    })

    // key, value pair to count number of orders per day

    let countVsDateObj = {};
    for(let i=0; ordersCopy1 && i<ordersCopy1.length; i++) {
        if(ordersCopy1[i]["created"][0] in countVsDateObj) {
            countVsDateObj[ordersCopy1[i]["created"][0]] += 1;
        
        } else {
            countVsDateObj[ordersCopy1[i]["created"][0]] = 1;
        }
    }
    
    // extracting orders' count and date from ordersCount object
    
    let ordersCount = Object.values(countVsDateObj);
    let date = Object.keys(countVsDateObj);

    return (
      <div>
        {ordersCount && date && (
          <BarGraph ordersCount={ordersCount} date={date} />
        )}
      </div>
    );
}

export default OredersCountVsDate
