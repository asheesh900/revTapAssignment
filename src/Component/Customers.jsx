import React, {useState} from 'react'
import db from '../Data/db.json'
import CustomersTable from './CustomersTable'
import Pagination from './Pagination';

function Customers() {
    let [customers] = useState(db.customers)

    const [currentPage, setCurrentPage] = useState(1);
    const [itemsPerPage] = useState(10);

    // Get current page

    const indexOfLastItem = currentPage * itemsPerPage;
    const indexOfFirstItem = indexOfLastItem - itemsPerPage;
    const currentItems = customers.slice(indexOfFirstItem, indexOfLastItem);

    // Change page

    const paginate = (pageNumber) => {
        setCurrentPage(pageNumber);
    }
    return (
      <div id = "customerTable">
        <CustomersTable customers={currentItems} />
        <Pagination
          itemsPerPage={itemsPerPage}
          totalItems={customers.length}
          paginate={paginate}
        />

      </div>
    );
}

export default Customers
