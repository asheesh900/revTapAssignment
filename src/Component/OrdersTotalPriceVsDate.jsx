import React, {useState} from 'react'
import db from '../Data/db.json'
import LineGraph from './LineGraph'

function OrdersTotalPriceVsDate() {
    let [orders] = useState(db.orders)

    let ordersCopy = [...orders]  // shallow copy

    // sorting orders based on date

    ordersCopy.sort((a, b) => {
        let dateA = a.created.toUpperCase();
        let dateB = b.created.toUpperCase();

        if(dateA < dateB) {
            return -1;
        } else if(dateA > dateB) {
            return 1;
        } else {
            return 0;
        }
    })

    let ordersCopy1 = JSON.parse(JSON.stringify(ordersCopy)) // detaching reference of its blueprint (deep copy)

    // extracting date in readable format

    if(ordersCopy1) {
        ordersCopy1 = ordersCopy1.map(ele => {
            ele["created"] = ele["created"].split("T");
            return ele
        })
    }

    // key, value pair to get the cost of orders per day

    let priceVsDateObj = {};
    for(let i=0; ordersCopy1 && i<ordersCopy1.length; i++) {
        if(ordersCopy1[i]["created"][0] in priceVsDateObj) {
            priceVsDateObj[ordersCopy1[i]["created"][0]] += Number(ordersCopy1[i]["price"]);
        
        } else {
            priceVsDateObj[ordersCopy1[i]["created"][0]] = Number(ordersCopy1[i]["price"]);
        }
    }

    // extracting total price per day and date from priceVsDateObj
    
    let totalPricePerDay = Object.values(priceVsDateObj);
    let date = Object.keys(priceVsDateObj);

    return (
        <div>
            {
                totalPricePerDay && date && <LineGraph totalPricePerDay = {totalPricePerDay} date = {date} />
            }
        </div>
    )
}

export default OrdersTotalPriceVsDate
