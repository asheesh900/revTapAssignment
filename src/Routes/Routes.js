import React from 'react'
import {Link, Route, Switch, BrowserRouter} from 'react-router-dom'
import Customers from '../Component/Customers'
import OrdersTotalPriceVsDate from '../Component/OrdersTotalPriceVsDate'
import OrdersCountVsDate from '../Component/OrdersCountVsDate'
import ErrorPage from '../Component/ErrorPage'

function Routes() {
    return (
        <BrowserRouter>
            <div className = "container-fluid  p-4 bg-dark text-white">
                <Link className = " m-2 p-2 text-white" to = "/">Customers</Link>
                <Link className = " m-2 p-2 text-white" to = "/ordersTotalPriceVsDate">Order's Total Price Vs Date</Link>
                <Link className = " m-2 p-2 text-white" to = "/ordersCountVsDate">Order's Count Vs Date</Link>
            </div>
            <Switch>
                <Route exact path = "/" component = {Customers} />
                <Route path = "/ordersTotalPriceVsDate" component = {OrdersTotalPriceVsDate} />
                <Route path = "/ordersCountVsDate" component = {OrdersCountVsDate} />
                <Route component = {ErrorPage} />
            </Switch>
        </BrowserRouter>
    )
}

export default Routes
