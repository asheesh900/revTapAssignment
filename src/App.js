import React from 'react';
import './App.css';
import Routes from './Routes/Routes'

function App() {
  return (
    <div className="topContainer">
      <Routes />
    </div>
  );
}

export default App;
