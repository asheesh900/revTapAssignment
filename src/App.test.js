import React from 'react';
import renderer from 'react-test-renderer'
import App from './App';
import Routes from './Routes/Routes'
import Customers from './Component/Customers';
import CustomersTable from './Component/CustomersTable'
import Pagination from './Component/Pagination'
import OrdersTotalPriceVsDate from './Component/OrdersTotalPriceVsDate'
import OrdersCountVsDate from './Component/OrdersCountVsDate'
import LineGraph from './Component/LineGraph'
import BarGrapth from './Component/BarGraph'

it('App component snapshot', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Routes component snapshot', () => {
  const tree = renderer
    .create(<Routes />)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Customers component snapshot', () => {
  const tree = renderer
    .create(<Customers />)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('CustomersTable component snapshot', () => {
  const tree = renderer
    .create(<CustomersTable />)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Pagination component snapshot', () => {
  const tree = renderer
    .create(<Pagination />)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('OrdersTotalPriceVsDate component snapshot', () => {
    const tree = renderer
      .create(<OrdersTotalPriceVsDate />)
      .toJSON();
      expect(tree).toMatchSnapshot();
});

it('OrdersCountVsDate component snapshot', () => {
  const tree = renderer
    .create(<OrdersCountVsDate />)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('LineGraph component snapshot', () => {
  const tree = renderer
    .create(<LineGraph />)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('BarGrapth component snapshot', () => {
  const tree = renderer
    .create(<BarGrapth />)
    .toJSON();
    expect(tree).toMatchSnapshot();
});