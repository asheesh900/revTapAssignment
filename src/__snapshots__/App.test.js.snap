// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot 1`] = `
<div
  className="topContainer"
>
  <div
    className="container-fluid  p-4 bg-dark text-white"
  >
    <a
      className=" m-2 p-2 text-white"
      href="/"
      onClick={[Function]}
    >
      Customers
    </a>
    <a
      className=" m-2 p-2 text-white"
      href="/ordersTotalPriceVsDate"
      onClick={[Function]}
    >
      Order's Total Price Vs Date
    </a>
    <a
      className=" m-2 p-2 text-white"
      href="/ordersCountVsDate"
      onClick={[Function]}
    >
      Order's Count Vs Date
    </a>
  </div>
  <div
    className="container my-4"
  >
    <h1
      className="text-center my-4"
    >
      Customers
    </h1>
    <table
      className="table table-striped  table-responsive"
    >
      <thead
        className="thead-dark"
      >
        <tr>
          <th
            scope="col"
          >
            Sr. No.
          </th>
          <th
            scope="col"
          >
            ID
          </th>
          <th
            scope="col"
          >
            First Name
          </th>
          <th
            scope="col"
          >
            Last Name
          </th>
          <th
            scope="col"
          >
            Email
          </th>
          <th
            scope="col"
          >
            Created_at
          </th>
          <th
            scope="col"
          >
            Orders
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th>
            1
             
          </th>
          <th>
            0b62a22a-51b5-4594-921a-e25a12ece0b8
             
          </th>
          <th>
            Dion
             
          </th>
          <th>
            Donnelly
             
          </th>
          <th>
            Carmella_Willms@gmail.com
             
          </th>
          <th>
            2020-04-23T21:35:28.635Z
             
          </th>
          <th>
            2
             
          </th>
        </tr>
        <tr>
          <th>
            2
             
          </th>
          <th>
            0fcbff30-5e8d-4ffb-949e-c7f1301c71e1
             
          </th>
          <th>
            Diego
             
          </th>
          <th>
            Hahn
             
          </th>
          <th>
            Fatima_Larson@yahoo.com
             
          </th>
          <th>
            2020-04-02T04:50:54.591Z
             
          </th>
          <th>
            0
             
          </th>
        </tr>
        <tr>
          <th>
            3
             
          </th>
          <th>
            621777e6-e862-474b-a2db-5812d848cedc
             
          </th>
          <th>
            Reinhold
             
          </th>
          <th>
            Ondricka
             
          </th>
          <th>
            Gus17@yahoo.com
             
          </th>
          <th>
            2020-04-04T03:29:24.579Z
             
          </th>
          <th>
            2
             
          </th>
        </tr>
        <tr>
          <th>
            4
             
          </th>
          <th>
            2b6bb67e-fb90-4eb6-af7b-e85945fe40ad
             
          </th>
          <th>
            Mary
             
          </th>
          <th>
            Bernhard
             
          </th>
          <th>
            Layne67@gmail.com
             
          </th>
          <th>
            2020-04-26T05:05:10.164Z
             
          </th>
          <th>
            0
             
          </th>
        </tr>
        <tr>
          <th>
            5
             
          </th>
          <th>
            6c51fb63-eae9-4b72-b365-7db31612a543
             
          </th>
          <th>
            Napoleon
             
          </th>
          <th>
            Beahan
             
          </th>
          <th>
            Nia.Streich17@yahoo.com
             
          </th>
          <th>
            2020-04-09T01:39:29.883Z
             
          </th>
          <th>
            1
             
          </th>
        </tr>
        <tr>
          <th>
            6
             
          </th>
          <th>
            72598b8a-33ab-48ce-8539-f4e2d3fd7ae5
             
          </th>
          <th>
            Joseph
             
          </th>
          <th>
            Jacobs
             
          </th>
          <th>
            Rozella22@yahoo.com
             
          </th>
          <th>
            2020-04-22T22:43:24.254Z
             
          </th>
          <th>
            1
             
          </th>
        </tr>
        <tr>
          <th>
            7
             
          </th>
          <th>
            eea3a83f-99c7-4434-b3f5-e52679f1f682
             
          </th>
          <th>
            Chase
             
          </th>
          <th>
            Ryan
             
          </th>
          <th>
            Arden20@hotmail.com
             
          </th>
          <th>
            2020-04-08T08:44:51.927Z
             
          </th>
          <th>
            0
             
          </th>
        </tr>
        <tr>
          <th>
            8
             
          </th>
          <th>
            07ae0e34-9dac-4c32-bf03-39f8e79f5f86
             
          </th>
          <th>
            Doyle
             
          </th>
          <th>
            Grant
             
          </th>
          <th>
            Ruthie_Glover30@hotmail.com
             
          </th>
          <th>
            2020-04-26T10:47:51.802Z
             
          </th>
          <th>
            1
             
          </th>
        </tr>
        <tr>
          <th>
            9
             
          </th>
          <th>
            21419e19-48c6-420a-af0f-a501b5cbbef4
             
          </th>
          <th>
            Dolly
             
          </th>
          <th>
            Schmeler
             
          </th>
          <th>
            Lori.Roob4@gmail.com
             
          </th>
          <th>
            2020-04-14T04:10:23.744Z
             
          </th>
          <th>
            2
             
          </th>
        </tr>
        <tr>
          <th>
            10
             
          </th>
          <th>
            d046c38d-4b7c-463e-b941-80ac0d43084c
             
          </th>
          <th>
            Maritza
             
          </th>
          <th>
            Kunde
             
          </th>
          <th>
            Demetris_Tromp@hotmail.com
             
          </th>
          <th>
            2020-04-20T20:27:36.295Z
             
          </th>
          <th>
            3
             
          </th>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="d-flex justify-content-center "
  >
    <ul
      className="pagination "
    >
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          1
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          2
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          3
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          4
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          5
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          6
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          7
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          8
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          9
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          10
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Customers component snapshot 1`] = `
Array [
  <div
    className="container my-4"
  >
    <h1
      className="text-center my-4"
    >
      Customers
    </h1>
    <table
      className="table table-striped  table-responsive"
    >
      <thead
        className="thead-dark"
      >
        <tr>
          <th
            scope="col"
          >
            Sr. No.
          </th>
          <th
            scope="col"
          >
            ID
          </th>
          <th
            scope="col"
          >
            First Name
          </th>
          <th
            scope="col"
          >
            Last Name
          </th>
          <th
            scope="col"
          >
            Email
          </th>
          <th
            scope="col"
          >
            Created_at
          </th>
          <th
            scope="col"
          >
            Orders
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th>
            1
             
          </th>
          <th>
            0b62a22a-51b5-4594-921a-e25a12ece0b8
             
          </th>
          <th>
            Dion
             
          </th>
          <th>
            Donnelly
             
          </th>
          <th>
            Carmella_Willms@gmail.com
             
          </th>
          <th>
            2020-04-23T21:35:28.635Z
             
          </th>
          <th>
            2
             
          </th>
        </tr>
        <tr>
          <th>
            2
             
          </th>
          <th>
            0fcbff30-5e8d-4ffb-949e-c7f1301c71e1
             
          </th>
          <th>
            Diego
             
          </th>
          <th>
            Hahn
             
          </th>
          <th>
            Fatima_Larson@yahoo.com
             
          </th>
          <th>
            2020-04-02T04:50:54.591Z
             
          </th>
          <th>
            0
             
          </th>
        </tr>
        <tr>
          <th>
            3
             
          </th>
          <th>
            621777e6-e862-474b-a2db-5812d848cedc
             
          </th>
          <th>
            Reinhold
             
          </th>
          <th>
            Ondricka
             
          </th>
          <th>
            Gus17@yahoo.com
             
          </th>
          <th>
            2020-04-04T03:29:24.579Z
             
          </th>
          <th>
            2
             
          </th>
        </tr>
        <tr>
          <th>
            4
             
          </th>
          <th>
            2b6bb67e-fb90-4eb6-af7b-e85945fe40ad
             
          </th>
          <th>
            Mary
             
          </th>
          <th>
            Bernhard
             
          </th>
          <th>
            Layne67@gmail.com
             
          </th>
          <th>
            2020-04-26T05:05:10.164Z
             
          </th>
          <th>
            0
             
          </th>
        </tr>
        <tr>
          <th>
            5
             
          </th>
          <th>
            6c51fb63-eae9-4b72-b365-7db31612a543
             
          </th>
          <th>
            Napoleon
             
          </th>
          <th>
            Beahan
             
          </th>
          <th>
            Nia.Streich17@yahoo.com
             
          </th>
          <th>
            2020-04-09T01:39:29.883Z
             
          </th>
          <th>
            1
             
          </th>
        </tr>
        <tr>
          <th>
            6
             
          </th>
          <th>
            72598b8a-33ab-48ce-8539-f4e2d3fd7ae5
             
          </th>
          <th>
            Joseph
             
          </th>
          <th>
            Jacobs
             
          </th>
          <th>
            Rozella22@yahoo.com
             
          </th>
          <th>
            2020-04-22T22:43:24.254Z
             
          </th>
          <th>
            1
             
          </th>
        </tr>
        <tr>
          <th>
            7
             
          </th>
          <th>
            eea3a83f-99c7-4434-b3f5-e52679f1f682
             
          </th>
          <th>
            Chase
             
          </th>
          <th>
            Ryan
             
          </th>
          <th>
            Arden20@hotmail.com
             
          </th>
          <th>
            2020-04-08T08:44:51.927Z
             
          </th>
          <th>
            0
             
          </th>
        </tr>
        <tr>
          <th>
            8
             
          </th>
          <th>
            07ae0e34-9dac-4c32-bf03-39f8e79f5f86
             
          </th>
          <th>
            Doyle
             
          </th>
          <th>
            Grant
             
          </th>
          <th>
            Ruthie_Glover30@hotmail.com
             
          </th>
          <th>
            2020-04-26T10:47:51.802Z
             
          </th>
          <th>
            1
             
          </th>
        </tr>
        <tr>
          <th>
            9
             
          </th>
          <th>
            21419e19-48c6-420a-af0f-a501b5cbbef4
             
          </th>
          <th>
            Dolly
             
          </th>
          <th>
            Schmeler
             
          </th>
          <th>
            Lori.Roob4@gmail.com
             
          </th>
          <th>
            2020-04-14T04:10:23.744Z
             
          </th>
          <th>
            2
             
          </th>
        </tr>
        <tr>
          <th>
            10
             
          </th>
          <th>
            d046c38d-4b7c-463e-b941-80ac0d43084c
             
          </th>
          <th>
            Maritza
             
          </th>
          <th>
            Kunde
             
          </th>
          <th>
            Demetris_Tromp@hotmail.com
             
          </th>
          <th>
            2020-04-20T20:27:36.295Z
             
          </th>
          <th>
            3
             
          </th>
        </tr>
      </tbody>
    </table>
  </div>,
  <div
    className="d-flex justify-content-center "
  >
    <ul
      className="pagination "
    >
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          1
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          2
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          3
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          4
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          5
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          6
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          7
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          8
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          9
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          10
        </button>
      </li>
    </ul>
  </div>,
]
`;

exports[`CustomersTable component snapshot 1`] = `
<div
  className="container my-4"
>
  <h1
    className="text-center my-4"
  >
    Customers
  </h1>
  <table
    className="table table-striped  table-responsive"
  >
    <thead
      className="thead-dark"
    >
      <tr>
        <th
          scope="col"
        >
          Sr. No.
        </th>
        <th
          scope="col"
        >
          ID
        </th>
        <th
          scope="col"
        >
          First Name
        </th>
        <th
          scope="col"
        >
          Last Name
        </th>
        <th
          scope="col"
        >
          Email
        </th>
        <th
          scope="col"
        >
          Created_at
        </th>
        <th
          scope="col"
        >
          Orders
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`Pagination component snapshot 1`] = `
<div
  className="d-flex justify-content-center "
>
  <ul
    className="pagination "
  />
</div>
`;

exports[`Routes component snapshot 1`] = `
Array [
  <div
    className="container-fluid  p-4 bg-dark text-white"
  >
    <a
      className=" m-2 p-2 text-white"
      href="/"
      onClick={[Function]}
    >
      Customers
    </a>
    <a
      className=" m-2 p-2 text-white"
      href="/ordersTotalPriceVsDate"
      onClick={[Function]}
    >
      Order's Total Price Vs Date
    </a>
    <a
      className=" m-2 p-2 text-white"
      href="/ordersCountVsDate"
      onClick={[Function]}
    >
      Order's Count Vs Date
    </a>
  </div>,
  <div
    className="container my-4"
  >
    <h1
      className="text-center my-4"
    >
      Customers
    </h1>
    <table
      className="table table-striped  table-responsive"
    >
      <thead
        className="thead-dark"
      >
        <tr>
          <th
            scope="col"
          >
            Sr. No.
          </th>
          <th
            scope="col"
          >
            ID
          </th>
          <th
            scope="col"
          >
            First Name
          </th>
          <th
            scope="col"
          >
            Last Name
          </th>
          <th
            scope="col"
          >
            Email
          </th>
          <th
            scope="col"
          >
            Created_at
          </th>
          <th
            scope="col"
          >
            Orders
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th>
            1
             
          </th>
          <th>
            0b62a22a-51b5-4594-921a-e25a12ece0b8
             
          </th>
          <th>
            Dion
             
          </th>
          <th>
            Donnelly
             
          </th>
          <th>
            Carmella_Willms@gmail.com
             
          </th>
          <th>
            2020-04-23T21:35:28.635Z
             
          </th>
          <th>
            2
             
          </th>
        </tr>
        <tr>
          <th>
            2
             
          </th>
          <th>
            0fcbff30-5e8d-4ffb-949e-c7f1301c71e1
             
          </th>
          <th>
            Diego
             
          </th>
          <th>
            Hahn
             
          </th>
          <th>
            Fatima_Larson@yahoo.com
             
          </th>
          <th>
            2020-04-02T04:50:54.591Z
             
          </th>
          <th>
            0
             
          </th>
        </tr>
        <tr>
          <th>
            3
             
          </th>
          <th>
            621777e6-e862-474b-a2db-5812d848cedc
             
          </th>
          <th>
            Reinhold
             
          </th>
          <th>
            Ondricka
             
          </th>
          <th>
            Gus17@yahoo.com
             
          </th>
          <th>
            2020-04-04T03:29:24.579Z
             
          </th>
          <th>
            2
             
          </th>
        </tr>
        <tr>
          <th>
            4
             
          </th>
          <th>
            2b6bb67e-fb90-4eb6-af7b-e85945fe40ad
             
          </th>
          <th>
            Mary
             
          </th>
          <th>
            Bernhard
             
          </th>
          <th>
            Layne67@gmail.com
             
          </th>
          <th>
            2020-04-26T05:05:10.164Z
             
          </th>
          <th>
            0
             
          </th>
        </tr>
        <tr>
          <th>
            5
             
          </th>
          <th>
            6c51fb63-eae9-4b72-b365-7db31612a543
             
          </th>
          <th>
            Napoleon
             
          </th>
          <th>
            Beahan
             
          </th>
          <th>
            Nia.Streich17@yahoo.com
             
          </th>
          <th>
            2020-04-09T01:39:29.883Z
             
          </th>
          <th>
            1
             
          </th>
        </tr>
        <tr>
          <th>
            6
             
          </th>
          <th>
            72598b8a-33ab-48ce-8539-f4e2d3fd7ae5
             
          </th>
          <th>
            Joseph
             
          </th>
          <th>
            Jacobs
             
          </th>
          <th>
            Rozella22@yahoo.com
             
          </th>
          <th>
            2020-04-22T22:43:24.254Z
             
          </th>
          <th>
            1
             
          </th>
        </tr>
        <tr>
          <th>
            7
             
          </th>
          <th>
            eea3a83f-99c7-4434-b3f5-e52679f1f682
             
          </th>
          <th>
            Chase
             
          </th>
          <th>
            Ryan
             
          </th>
          <th>
            Arden20@hotmail.com
             
          </th>
          <th>
            2020-04-08T08:44:51.927Z
             
          </th>
          <th>
            0
             
          </th>
        </tr>
        <tr>
          <th>
            8
             
          </th>
          <th>
            07ae0e34-9dac-4c32-bf03-39f8e79f5f86
             
          </th>
          <th>
            Doyle
             
          </th>
          <th>
            Grant
             
          </th>
          <th>
            Ruthie_Glover30@hotmail.com
             
          </th>
          <th>
            2020-04-26T10:47:51.802Z
             
          </th>
          <th>
            1
             
          </th>
        </tr>
        <tr>
          <th>
            9
             
          </th>
          <th>
            21419e19-48c6-420a-af0f-a501b5cbbef4
             
          </th>
          <th>
            Dolly
             
          </th>
          <th>
            Schmeler
             
          </th>
          <th>
            Lori.Roob4@gmail.com
             
          </th>
          <th>
            2020-04-14T04:10:23.744Z
             
          </th>
          <th>
            2
             
          </th>
        </tr>
        <tr>
          <th>
            10
             
          </th>
          <th>
            d046c38d-4b7c-463e-b941-80ac0d43084c
             
          </th>
          <th>
            Maritza
             
          </th>
          <th>
            Kunde
             
          </th>
          <th>
            Demetris_Tromp@hotmail.com
             
          </th>
          <th>
            2020-04-20T20:27:36.295Z
             
          </th>
          <th>
            3
             
          </th>
        </tr>
      </tbody>
    </table>
  </div>,
  <div
    className="d-flex justify-content-center "
  >
    <ul
      className="pagination "
    >
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          1
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          2
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          3
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          4
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          5
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          6
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          7
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          8
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          9
        </button>
      </li>
      <li
        className="page-item"
      >
        <button
          className="page-link"
          onClick={[Function]}
        >
          10
        </button>
      </li>
    </ul>
  </div>,
]
`;
